lib/gooddata/extensions/hash.rb

```diff
@@ -1,48 +1,17 @@
-# encoding: UTF-8
-#
 # Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-class Hash
-  # Return a hash that includes everything but the given keys. This is useful for
-  # limiting a set of parameters to everything but a few known toggles:
-  #
-  #   @person.update_attributes(params[:person].except(:admin))
-  #
-  # If the receiver responds to +convert_key+, the method is called on each of the
-  # arguments. This allows +except+ to play nice with hashes with indifferent access
-  # for instance:
-  #
-  #   {:a => 1}.with_indifferent_access.except(:a)  # => {}
-  #   {:a => 1}.with_indifferent_access.except("a") # => {}
-  #
-  def except(*keys)
-    dup.except!(*keys)
-  end
-
-  # Replaces the hash without the given keys.
-  def except!(*keys)
-    keys.each { |key| delete(key) }
-    self
-  end
-
-  def slice(*keys)
-    keys.map! { |key| convert_key(key) } if respond_to?(:convert_key, true)
-    keys.each_with_object(self.class.new) { |k, hash| hash[k] = self[k] if key?(k) }
-  end
-
-  def compact
-    select { |_, value| !value.nil? }
-  end
-
-  def deep_merge(hash)
-    hash = hash.to_hash
-    merge(hash) do |_key, old_val, new_val|
-      if old_val.is_a?(Hash) && new_val.is_a?(Hash)
-        old_val.deep_merge(new_val)
-      else
-        new_val
+module HashExtensions
+  refine Hash do
+    def deep_merge(hash)
+      hash = hash.to_hash
+      merge(hash) do |_key, old_val, new_val|
+        if old_val.is_a?(Hash) && new_val.is_a?(Hash)
```

Comment:
info: Refers to 'old_val' more than self (maybe move it to another class?) (FeatureEnvy) (reek)